### PR TITLE
fix: [CDS-78795]: missing conditional execution icon for stage and step templates with runtime conditional execution

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Whenever a stage / step has conditional execution configured for the entity we show the conditional execution icon on the entity node
- But if the similar entity references a template which has conditional execution configured the icon was missing
- As received in the recent feedback from Regions bank we added the extra condition to check the presence of when condition in template as well and show the concerned icon

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/dd60e48e-8689-4707-b9ec-ecdd712d0d1d

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineDiagram/PipelineGraph/PipelineGraphUtils.ts [hunk 1]
```diff
@@ -491,6 +491,19 @@ const getPipelineGraphData = ({
   return graphState
 }
 
+function getIsConditionalExecutionEnabled(entitiy: any): boolean {
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineDiagram/PipelineGraph/PipelineGraphUtils.ts:494]:
```suggestion
function getIsConditionalExecutionEnabled(entity: any): boolean {
```

File: src/modules/70-pipeline/components/PipelineDiagram/PipelineGraph/PipelineGraphUtils.ts [hunk 2]
```diff
@@ -538,9 +551,7 @@ const transformStageData = ({
           }),
           isInComplete: isCustomGeneratedString(identifier) || hasErrors,
           loopingStrategyEnabled: !!stage.stage?.strategy,
-          conditionalExecutionEnabled: stage.stage.when
-            ? stage.stage.when?.pipelineStatus !== 'Success' || !!stage.stage.when?.condition?.trim()
-            : false,
+          conditionalExecutionEnabled: getIsConditionalExecutionEnabled(stage.stage),
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineDiagram/PipelineGraph/PipelineGraphUtils.ts:554]: Add similar conditions for parallel node and else case L589,L632

File: src/modules/70-pipeline/components/PipelineDiagram/PipelineGraph/PipelineGraphUtils.ts (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineDiagram/PipelineGraph/PipelineGraphUtils.ts]: done

File: src/modules/70-pipeline/components/PipelineDiagram/PipelineGraph/PipelineGraphUtils.ts (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/70-pipeline/components/PipelineDiagram/PipelineGraph/PipelineGraphUtils.ts:554]: done

**Review by PratyushGar — APPROVED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request merged**